gdnative-core/src/variant_array.rs

```diff
@@ -171,76 +156,224 @@ impl VariantArray {
+        unsafe { (get_api().godot_array_erase)(self.sys_mut(), val.sys()) }
+    }
+
+    #[inline]
+    pub fn resize(&self, size: i32) {
+        unsafe { (get_api().godot_array_resize)(self.sys_mut(), size) }
+    }
+
+    /// Appends an element at the end of the array.
+    #[inline]
+    pub fn push(&self, val: &Variant) {
+        unsafe {
+            (get_api().godot_array_push_back)(self.sys_mut(), val.sys());
+        }
+    }
+
+    /// Removes an element at the end of the array.
+    #[inline]
+    pub fn pop(&self) -> Variant {
+        unsafe { Variant((get_api().godot_array_pop_back)(self.sys_mut())) }
+    }
+
+    /// Appends an element to the front of the array.
+    #[inline]
+    pub fn push_front(&self, val: &Variant) {
+        unsafe {
+            (get_api().godot_array_push_front)(self.sys_mut(), val.sys());
+        }
+    }
+
+    /// Removes an element at the front of the array.
+    #[inline]
+    pub fn pop_front(&self) -> Variant {
+        unsafe { Variant((get_api().godot_array_pop_front)(self.sys_mut())) }
+    }
+
+    /// Insert a new int at a given position in the array.
+    #[inline]
+    pub fn insert(&self, at: i32, val: &Variant) {
+        unsafe { (get_api().godot_array_insert)(self.sys_mut(), at, val.sys()) }
+    }
+
+    /// Returns an iterator through all values in the `VariantArray`.
+    ///
+    /// `VariantArray` is reference-counted and have interior mutability in Rust parlance.
+    /// Modifying the same underlying collection while observing the safety assumptions will
+    /// not violate memory safely, but may lead to surprising behavior in the iterator.
     #[inline]
-    pub fn iter(&self) -> Iter {
-        Iter {
+    pub fn iter(&self) -> IterUnique {
+        IterUnique {
             arr: self,
             range: 0..self.len(),
         }
     }
 
+    /// Returns an iterator through all values in the `VariantArray`.
+    ///
+    /// `VariantArray` is reference-counted and have interior mutability in Rust parlance.
+    /// Modifying the same underlying collection while observing the safety assumptions will
+    /// not violate memory safely, but may lead to surprising behavior in the iterator.
     #[inline]
-    pub fn iter_mut(&mut self) -> IterMut {
+    pub fn into_iter(self) -> IntoIterUnique {
         let len = self.len();
-        IterMut {
+        IntoIterUnique {
             arr: self,
             range: 0..len,
         }
     }
+}
 
-    #[doc(hidden)]
+impl VariantArray<Shared> {
     #[inline]
-    pub fn sys(&self) -> *const sys::godot_array {
-        &self.0
+    pub fn new_shared() -> Self {
+        VariantArray::<Unique>::new().into_shared()
     }
 
-    #[doc(hidden)]
     #[inline]
-    pub fn from_sys(sys: sys::godot_array) -> Self {
-        VariantArray(sys)
+    pub unsafe fn assume_unique(self) -> VariantArray<Unique> {
+        VariantArray::<Unique> {
+            sys: self.sys,
+            _marker: PhantomData,
+        }
+    }
```

Comment:
`self` is dropped here, but the reference isn't incremented because the constructor is used directly.